lab-yunjoo/app/controller/game-controller.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+// npm modules
+const angular = require('angular');
+// app modules
+const Monster = require('../model/monster');
+
+// module logic
+angular.module('demoApp').controller('GameController', [GameController]);
+
+function GameController(){
+  this.history = ['WELCOME TO THE SALTMINE!'];
+  this.moveCount = 0;
+  this.map = require('../model/map');
+  this.oldLocation = '';
+  this.newLocation = '';
+  this.player = {
+    name: 'amazingYun',
+    hp: 100,
+    dammage: 10,
+    xp: 0,
+    location: 'roomA'
+  };
+
+  this.gameForm = {
+    direction: 'north'
+  };
+
+  this.room = {
+    name: this.player.location
+  };
+}
+
+GameController.prototype.moveDirection = function(direction){
+  this.moveCount++;
+  this.oldLocation = this.player.location;
+  this.newLocation = this.map[this.oldLocation][direction];
+  if (this.newLocation && this.newLocation !== 'wall'){
+    // move into that location
+    this.updateLocation(this.newLocation);
+    return;
+  }
+
+  // stay were you are
+  this.holdLocation();
+};
+
+GameController.prototype.updateLocation = function(location){
+  this.player.location = location;
+  this.room.name = location;
+  if (Math.random() < this.map[location].monsterChance){
+    this.room.monster = new Monster();
+    this.player.hp -= this.room.monster.dammage;
+    this.logTurn(`is now in ${this.player.location}. a ${this.room.monster.name} attacked. you lost ${this.room.monster.dammage}`);
```

Comment:
I had an issue where it told me that 'a undefined attacked.' Something about the name isn't working quite right.